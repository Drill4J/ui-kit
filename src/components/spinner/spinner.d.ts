interface Props {
    className?: string;
    disabled?: boolean;
}
export declare const Spinner: {
    ({ className, disabled, }: Props): JSX.Element;
    displayName: string;
    __docgenInfo: {
        description: string;
        displayName: string;
        props: {
            className: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
            disabled: {
                defaultValue: null;
                description: string;
                name: string;
                required: boolean;
                type: {
                    name: string;
                };
            };
        };
    };
};
export {};
