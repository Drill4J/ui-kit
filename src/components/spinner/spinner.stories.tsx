import React from 'react';
import { ComponentMeta } from '@storybook/react';
import 'twin.macro';

import { Spinner } from './spinner';

export default {
  title: 'Spinner',
  component: Spinner,
} as ComponentMeta<typeof Spinner>;

const Template = (args) => <Spinner {...args} />;
export const Default = Template.bind({});

export const Disabled = Template.bind({});
Disabled.args = { disabled: true };
