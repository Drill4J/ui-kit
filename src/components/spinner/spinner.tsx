import React from 'react';
import tw, { styled, css } from 'twin.macro';

export const Spinner = styled.div<{color?: 'blue' | 'white'}>(({ color }) => [
  tw`animate-spin w-4 h-4 rounded-full`,
  css`
    border: solid rgba(255, 255, 255, 0.15) 4px;
    border-top:solid rgb(255, 255, 255) 4px;`,
  color === 'blue' && css`
    border: solid rgba(0, 127, 255, 0.15) 4px;
    border-top:solid rgb(0, 127, 255) 4px;`,
]);
