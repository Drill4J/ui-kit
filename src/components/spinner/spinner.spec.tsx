import React from 'react';
import renderer from 'react-test-renderer';
import { Spinner } from './spinner';

describe('Badge', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Spinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
