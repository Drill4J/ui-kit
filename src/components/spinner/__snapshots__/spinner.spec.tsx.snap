// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<div
  className="spinner__Wrapper-sc-gvv5fx-0 hnGMMU"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
